content/blog/2018/delayering-istio/delayering-istio.md

```diff
@@ -0,0 +1,216 @@
+---
+title: Delayering Istio with AppSwitch
+description: Automatic application onboarding and latency optimizations by removing unnecessary layers with AppSwitch
+publishdate: 2018-06-28
+subtitle:
+attribution: Dinesh Subhraveti
+weight: 90
+keywords: [performance, traffic-management, appswitch]
+---
+
+>_All problems in computer science can be solved with another layer, except of course the problem of too many layers._ -- David Wheeler
+
+The sidecar proxy approach enables a lot of awesomeness.  Squarely in the datapath between microservices, the sidecar can precisely tell what the application is trying to do.  It can monitor and instrument protocol traffic, not in the bowels of the networking layers but at the application level, to enable deep visibility, access controls and traffic management.
+
+If we look closely however, there are many intermediate layers that the data has to pass through before the high-value analysis of application-traffic can be performed.  Most of those layers are part of the base plumbing infrastructure that are there just to push the data along.  In so doing, they add latency to communication and complexity to the overall system.
+
+Over the years, there has been much collective effort in implementing aggressive fine-grain optimizations within the layers of the network datapath.  Each iteration may shave another few microseconds.  But then the true necessity of those layers itself has not been questioned.
+
+## Don’t optimize layers, remove them
+
+In my belief, optimizing something is a poor fallback to removing its requirement altogether.  That was the goal of my [initial work](https://apporbit.com/a-brief-history-of-containers-from-reality-to-hype) on OS-level virtualization that led to Linux containers which effectively [removed virtual machines](https://www.oreilly.com/ideas/the-unwelcome-guest-why-vms-arent-the-solution-for-next-gen-applications) by running applications directly on the host operating system without requiring an intermediate guest.  For a long time the industry was fighting the wrong battle distracted by optimizing VMs rather than removing the additional layer altogether.
+
+I see the same pattern repeat itself with the connectivity of microservices, and networking in general.  Network has been going through the changes that physical servers have gone through a decade earlier.  New set of layers and constructs are being introduced.  They are being baked deep into the protocol stack and even silicon without adequately considering low-touch alternatives.  Perhaps there is a way to remove those additional layers altogether.
+
+{{< image width="25%" ratio="175%"
+    link="burger.jpg"
+    alt="What current network stack has become"
+    >}}
```

Comment:
while this is an opinion/tech article, this picture is more suitable to a presentation. I am not sure this will fit the rest of the blogs in the Istio site, that are more matter of fact figures about the architecture, etc.
Thoughts @geeknoid ?